ksqldb-functional-tests/src/test/resources/query-validation-tests/formats.json

```diff
@@ -17,6 +17,265 @@
       },
       "inputs": [],
       "outputs": []
+    },
+    {
+      "name": "missing value format",
+      "statements": [
+        "CREATE STREAM TEST WITH (kafka_topic='test_topic');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "JSON"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.parser.exception.ParseFailedException",
+        "message": "Failed to prepare statement: Missing required property \"VALUE_FORMAT\" which has no default value."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "key format without feature flag",
+      "statements": [
+        "CREATE STREAM TEST (foo VARCHAR) WITH (kafka_topic='test_topic', value_format='JSON', key_format='KAFKA');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "JSON"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The use of 'KEY_FORMAT' and 'FORMAT' is disabled, as this feature is under development."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "format without feature flag",
+      "statements": [
+        "CREATE STREAM TEST (foo VARCHAR) WITH (kafka_topic='test_topic', format='KAFKA');"
+      ],
+      "topics": [
+        {
+          "name": "test_topic",
+          "format": "KAFKA"
+        }
+      ],
+      "expectedException": {
+        "type": "io.confluent.ksql.util.KsqlStatementException",
+        "message": "The use of 'KEY_FORMAT' and 'FORMAT' is disabled, as this feature is under development."
+      },
+      "inputs": [],
+      "outputs": []
+    },
+    {
+      "name": "explicit key and optional value format",
+      "statements": [
+        "CREATE STREAM INPUT (K STRING KEY, foo INT) WITH (kafka_topic='input_topic', key_format='KAFKA');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM INPUT;"
+      ],
+      "properties": {
+        "ksql.persistence.default.format.value": "JSON",
+        "ksql.key.format.enabled": true
+      },
+      "inputs": [
+        {"topic": "input_topic", "value": {"FOO": 10}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": {"FOO": 10}}
+      ]
+    },
+    {
+      "name": "optional key and value format",
+      "statements": [
+        "CREATE STREAM INPUT (K STRING KEY, foo INT) WITH (kafka_topic='input_topic');",
+        "CREATE STREAM OUTPUT AS SELECT * FROM INPUT;"
+      ],
+      "properties": {
+        "ksql.persistence.default.format.value": "JSON",
+        "ksql.key.format.enabled": true
+      },
+      "inputs": [
+        {"topic": "input_topic", "value": {"FOO": 10}}
+      ],
+      "outputs": [
+        {"topic": "OUTPUT", "value": {"FOO": 10}}
+      ]
+    },
+    {
+      "name": "unsupported key format",
```

Comment:
This test (and also "unsupported format" below) currently does not pass in QTT even though it works if you actually spin up a server. This is because QTT does not run the RequestValidator, which is where the check is currently performed. I looked into adding the RequestValidator into QTT but it's not clear to me whether this is possible / the correct thing to do.